src/main/java/org/fogbowcloud/arrebol/ArrebolController.java

```diff
@@ -62,54 +73,39 @@ public ArrebolController(Properties properties) {
         this.queue = new TaskQueue(queueId, queueName);
 
         int poolId = 1;
-        WorkerPool pool = createPool(arrebolProperties, poolId);
+        WorkerPool pool = createPool(poolId);
 
         //create the scheduler bind the pieces together
         FifoSchedulerPolicy policy = new FifoSchedulerPolicy();
         this.scheduler = new DefaultScheduler(queue, pool, policy);
 
-        this.jobPool = Collections.synchronizedMap(new HashMap<String,  Job>());
+        this.jobPool = Collections.synchronizedMap(new HashMap<String, Job>());
         this.jobDatabaseCommitter = new Timer(true);
+        this.jobStateMonitor = new Timer(true);
     }
 
-    private static final String RAW_TYPE = "raw";
-    private static final String DOCKER_TYPE = "docker";
 
-    private WorkerPool createPool(Properties properties, int poolId) {
 
-        //we need to deal with missing/wrong properties
+    private Configuration loadConfigurationFile(String path) throws FileNotFoundException {
+        Configuration configuration;
+        Gson gson = new Gson();
+        BufferedReader bufferedReader = new BufferedReader(new FileReader(path + File.separator + "arrebol.json"));
+        configuration = gson.fromJson(bufferedReader, Configuration.class);
+        return configuration;
+    }
 
-        Collection<Worker> workers = new LinkedList<>();
-        String poolType = properties.getProperty("pool.type");
+    private WorkerPool createPool(int poolId) {
 
-        int poolSize = new Integer(properties.getProperty("pool.size"));
-        for (int i = 0; i < poolSize; i++) {
-            TaskExecutor executor = createTaskExecutor(poolType, properties);
-            Specification resourceSpec = null;
-            Worker worker = new MatchAnyWorker(resourceSpec, "resourceId-"+i, poolId, executor);
-            workers.add(worker);
-        }
+        //we need to deal with missing/wrong properties
```

Comment:
We still not dealt with missing properties?

Reply:
Inside each specific worker configuration has a validation. However, didn't have a verification about pool type. Added now as Conf Validator.